examples/calculator/lib/calc_expression.dart

```diff
@@ -0,0 +1,340 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// A token that composes an expression. There are several kinds of tokens
+/// that represent arithmetic operation symbols, numbers and pieces of numbers.
+/// We need to represent pieces of numbers because the user may have only
+/// entered a partial expression so far.
+class ExpressionToken {
+  final String stringRep;
+
+  ExpressionToken(this.stringRep);
+
+  @override
+  String toString() => stringRep;
+}
+
+/// A token that represents a number.
+class NumberToken extends ExpressionToken {
```

Comment:
...although it looks like you have a number of subclasses that use both, so maybe not...

In general I would avoid using `num`, because it leads to polymorphic call sites, but performance isn't critical here so whatever.